Finished cesium load error.

teamagam3 opened the pull request:
Cesium now reloads every time the map failes to load.

teamagam3 committed:
Cesium map reloads when it fails

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/49" target="_blank"><img src="https://app.review.ninja/52010471/pull/49/badge" alt="ReviewNinja"/>

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -264,6 +270,15 @@ public boolean isReady() {
+        if(!mIsHandlingError.getData()) {
+            mIsHandlingError.setData(true);
+            init();
+            loadUrl(FILE_ANDROID_ASSET_VIEWER);
```
why do you load the url second time?
you have the same line at the end of init !fix

teamagam1 replied:
!resolved

teamagam1 replied:
!resolved

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:
```diff
@@ -0,0 +1,60 @@
+
+
+    @Override
+    public boolean onConsoleMessage(ConsoleMessage consoleMessage) {
```
are you sure that consoleMessage is not null and has all the fields?

gRaytan replied:
whats is the answer? what about try catch?

teamagam1 replied:
!completed

teamagam1 replied:
yes we are sure

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:
```diff
@@ -0,0 +1,60 @@
+        // A listener should be called.
+        if (fileSource.toLowerCase().contains(CESIUM_LIBRARY_FILE)) {
+            if (message.toLowerCase().contains(CESIUM_ERROR_MESSAGE)) {
+                return true;
```
that's the message? no details?

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -264,6 +270,15 @@ public boolean isReady() {
     }
 
+    @Override
+    public void onError(String error) {
```
give it more detailed name. !resolve

teamagam1 replied:
!resolved

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -264,6 +270,15 @@ public boolean isReady() {
+    public void onError(String error) {
+        if(!mIsHandlingError.getData()) {
+            mIsHandlingError.setData(true);
+            init();
```
this event-loop is very heavy and it also run in the uithread which is not acceptable.

teamagam1 replied:
!resolve

teamagam1 replied:
!resolved

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -273,10 +288,15 @@ public boolean isReady() {
                 public void run() {
+                    CesiumMapView.this.mIsHandlingError.setData(false);
+
                     CesiumMapView.this.mIsGGMapReadySynchronized.setData(true);
```
can you use mIsGGMapReadySynchronized for this? this is not critical...

teamagam1 replied:
!resolved  !resolved

teamagam1 commented:
check

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -28,7 +28,9 @@
  * Wrapper view class for a WebView-based Cesium viewer
  */
-public class CesiumMapView extends WebView implements GGMapView, VectorLayer.LayerChangedListener {
+public class CesiumMapView
```
why why why???

# @

[](url)

`public static void main(){
}`

> if you want to shoot shoot dont talk
- [x] complete
- [x] resolve
- דכגדכ

teamagam1 replied:
!completed

teamagam1 commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -28,7 +28,9 @@
+        extends WebView
+        implements GGMapView, VectorLayer.LayerChangedListener, CesiumWebChromeClient.CesiumJsErrorListener {
 
     public static final String FILE_ANDROID_ASSET_VIEWER =
```
clean

teamagam1 replied:
!completed

gRaytan committed:
Merge remote-tracking branch 'origin/develop' into task/map_related_bugs/cesium_load_error
iver

(cherry picked from commit c84442449662ee3808a1eec3fad506e915344494)

# Conflicts:
#	app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/ConnectivityStatusReceiver.java
#	app/src/main/java/com/teamagam/gimelgimel/app/network/receivers/NetworkChangeReceiver.java
#	app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGImageService.java
#	app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGMessagingUtils.java
#	app/src/main/java/com/teamagam/gimelgimel/app/network/services/message_polling/MessagePoller.java
#	app/src/main/java/com/teamagam/gimelgimel/app/view/MainActivity.java

# Conflicts:
#	app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java
#	app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java

teamagam1 committed:
using network broadcast receiver to reload map (imageryProvider).

the CesiumMapView calls js to remove the bad layer and replace it with layer that connects to the internet.

teamagam1 committed:
Merge remote-tracking branch 'origin/feature/map_related_bugs' into task/map_related_bugs/cesium_load_error

# Conflicts:
#	app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java

gRaytan committed:
Merge branch 'feature/map_related_bugs' into task/map_related_bugs/cesium_load_error

teamagam1 commented:
@agamgil יאללה

gRaytan commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumBaseBridge.java:
```diff
@@ -18,6 +18,8 @@ public CesiumBaseBridge(JavascriptCommandExecutor javascriptCommandExecutor) {
         mJsExecutor = javascriptCommandExecutor;
     }
 
+
```
no need for spalce, delete

gRaytan replied:
igone it

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumWebChromeClient.java:
```diff
@@ -0,0 +1,60 @@
+        String message = consoleMessage.message();
+
+        if (shouldRaiseEvent(fileSource, message)) {
+            RaiseEvent(message);
```
camelCase !fix

teamagam1 replied:
!fixed

gRaytan commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -273,6 +302,21 @@ public boolean isReady() {
     @Override
+    public void onCesiumError(String error) {
+        if (error.contains("ImageryProvider")) {
+            mIsHandlingError.setData(true);
```
arent you suppose to insert data? why do we update with TRUE?

teamagam1 replied:
כי עכשיו אנחנו יודעים שיש בעיה. אז אנחנו זוכרים את זה במידע

teamagam1 committed:
more CR fixes
